File: src/main/java/com/logosprog/mainutils/files/properties/builders/PropertiesBuilder.java
/*
 * Copyright (c) 2015. This code is a LogosProg property. All Rights Reserved.
 */

package com.logosprog.mainutils.files.properties.builders;

import com.logosprog.mainutils.files.builders.ObjectFromFileBuilder;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

/**
 * Created by forando on 25.06.15.<br>
 * Builds {@link Properties} object from resource file.
 */
public class PropertiesBuilder implements ObjectFromFileBuilder<Properties, InputStream> {

    private Properties properties;

    public PropertiesBuilder(InputStream is) throws IOException {
        this.properties = this.build(is);
    }

    @Override
    public Properties build(InputStream is) throws IOException {
        Properties properties = new Properties();
        properties.load(is);
        return properties;
    }


    public Properties getProperties(){
        return properties;
    }
}

File: src/main/kotlin/com/logosprog/mainutils/files/properties/builders/PropertiesBuilder.kt
/*
 * Copyright (c) 2016. This code is a LogosProg property. All Rights Reserved.
 */

/*
 * Copyright (c) 2016. This code is a LogosProg property. All Rights Reserved.
 */

@file:Suppress("PARAMETER_NAME_CHANGED_ON_OVERRIDE", "unused")

package com.logosprog.mainutils.files.properties.builders

import com.logosprog.mainutils.files.builders.ObjectFromFileBuilder
import java.io.IOException
import java.io.InputStream
import java.util.*

/**
 * Builds [Properties] object from resource file.
 *
 * Created by forando on 25.06.15.
 */
open class PropertiesBuilder @Throws(IOException::class)
constructor(inputStream: InputStream) : ObjectFromFileBuilder<Properties, InputStream> {

    private val properties: Properties

    init {
        this.properties = this.build(inputStream)
    }

    @Throws(IOException::class)
    override fun build(inputStream: InputStream): Properties {
        val properties = Properties()
        properties.load(inputStream)
        return properties
    }
}
